server/app/services/program/ProgramService.java

```diff
@@ -814,12 +840,28 @@ public ErrorAnd<ProgramDefinition, CiviFormError> appendStatus(
                 .add(status)
                 .build());
 
-    return ErrorAnd.of(
-        syncProgramDefinitionQuestions(
-                programRepository.getShallowProgramDefinition(
-                    programRepository.updateProgramSync(program.toProgram())))
-            .toCompletableFuture()
-            .join());
+    ErrorAnd<ProgramDefinition, CiviFormError> result =
+        ErrorAnd.of(
+            syncProgramDefinitionQuestions(
+                    programRepository.getShallowProgramDefinition(
+                        programRepository.updateProgramSync(program.toProgram())))
+                .toCompletableFuture()
+                .join());
+    if (!result.isError()) {
+      // Temporary hook to create a new row in ApplicationStatuses table for the status change.
```

Comment:
**question:** Does it create in this case? If it only updates will you reflect that in the comment or if it can do either, can you add that as well.